### PR TITLE
Bugfix and Cleanup code

### PR DESCRIPTION
**Pull request by hskim881028**
SSIA

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Module/Inventory.cs [hunk 1]
```diff
@@ -502,45 +503,40 @@ private void UpdateDimmedInventoryItem()
...
                 : null;
-            Set(
-                itemSetDimPredicates: predicateList,
-                isArena: isArena,
-                useConsumable: useConsumable,
-                onUpdateInventory: onUpdateInventory);
-            _allowMoveTab = true;
+            SetInventoryTab(predicateList, inventoryType, inventory, onUpdateInventory, useConsumable);
+            _toggleGroup.DisabledFunc = () => false;
+            // _allowMoveTab = true;
```
Comment (ipdae) [inline nekoyume/Assets/_Scripts/UI/Module/Inventory.cs:520]:
```suggestion
```

File: nekoyume/Assets/_Scripts/UI/Module/Inventory.cs [hunk 2]
```diff
@@ -502,45 +503,40 @@ private void UpdateDimmedInventoryItem()
...
         public void SetShop(Action<InventoryItem, RectTransform> clickItem)
         {
-            _reverseOrder = false;
             _checkTradable = true;
             SetAction(clickItem);
-            Set();
-            _allowMoveTab = true;
+            SetInventoryTab();
+            _toggleGroup.DisabledFunc = () => false;
+            // _allowMoveTab = true;
```
Comment (ipdae) [inline nekoyume/Assets/_Scripts/UI/Module/Inventory.cs:529]:
```suggestion
```

File: nekoyume/Assets/_Scripts/UI/Module/Inventory.cs [hunk 3]
```diff
@@ -148,26 +143,32 @@ private void OnTabButtonClick(IToggleable toggleable, ItemType type, System.Acti
             itemSetDimPredicates?.ForEach(tuple =>
                 _dimConditionFuncsByItemType[tuple.type].Add(tuple.predicate));
 
-            if (isArena)
+            switch (inventoryType)
             {
-                var inventory = RxProps.PlayersArenaParticipant.Value.AvatarState.inventory;
-                SetInventory(inventory, onUpdateInventory);
+                case InventoryType.Default:
+                    ReactiveAvatarState.Inventory
```
Comment (ipdae) [inline nekoyume/Assets/_Scripts/UI/Module/Inventory.cs:149]: First of all, apologize for understanding of this code. could you tell me why subscribe only at this time?

**Review by sonohoshi — APPROVED** [review]

File: nekoyume/Assets/_Scripts/UI/Module/Inventory.cs (hunk 3 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/UI/Module/Inventory.cs:149]: In other cases, we are looking at an inventory other than the existing inventory in the avatar State.

Comment (hskim881028) [reply nekoyume/Assets/_Scripts/UI/Module/Inventory.cs:149]: > First of all, apologize for understanding of this code. could you tell me why subscribe only at this time?

https://github.com/planetarium/NineChronicles/pull/2051#discussion_r996544540

**Review by Unengine — APPROVED** [review]

**Review by boscohyun — APPROVED** [review]

**Pull request merged**